File: ios/WhoWouldWin/Models/BattleResult.swift
import Foundation

struct BattleResult: Codable, Equatable {
    let winner: String           // Animal ID of winner, or "draw"
    let narration: String        // 2-sentence battle narration
    let funFact: String          // One fun fact about the winner
    let winnerHealthPercent: Int // 10–90, how dominant the win was
    let loserHealthPercent: Int  // 0–25, how much fight the loser put up
    /// Optional one-line, kid-friendly reason the winner won (specific weapon/
    /// ability/size). Supplied by the cloud backend; nil for older/offline
    /// results. The result screen prefers this over the local BattleInsight.why.
    var why: String? = nil
    var isOfflineFallback: Bool = false  // true when using local fallback result

    enum CodingKeys: String, CodingKey {
        case winner, narration, funFact, winnerHealthPercent, loserHealthPercent, why
    }
}

File: android/app/src/main/java/com/whowouldin/whowouldwin/model/BattleResult.kt
package com.whowouldin.whowouldwin.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class BattleResult(
    val winner: String,               // Animal id, or "draw"
    val narration: String,            // 2-sentence battle narration
    val funFact: String,              // one fun fact about the winner
    val winnerHealthPercent: Int,     // 10–90 — how dominant the win was
    val loserHealthPercent: Int,      // 0–25 — how much fight the loser put up
    @Transient val isOfflineFallback: Boolean = false,
)
